toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt

```diff
@@ -125,24 +125,32 @@ class CratesLocalIndexServiceImpl
         this.state = state
     }
 
+    @Throws(CratesLocalIndexException::class)
     override fun getCrate(crateName: String): CargoRegistryCrate? {
-        if (isUpdating.get() || crates == null) return null
+        when {
```

Comment:
Let's replace `when` with two `if`s. This allows Kotlin compiler to infer that `crates` is  not null